crates/next-core/src/router_source.rs

```diff
@@ -60,6 +66,17 @@ impl ContentSource for NextRouterContentSource {
     ) -> Result<ContentSourceResultVc> {
         let this = self_vc.await?;
 
+        // The next-dev server can currently run against projects as simple as
+        // `index.js`. If this isn't a Next.js project, don't try to use the Next.js
+        // router.
+        let project_root = this.execution_context.await?.project_root;
+        let find_config_result = &*find_context_file(project_root, next_configs()).await?;
+        if matches!(find_config_result, FindContextFileResult::NotFound(_)) {
```

Comment:
Nit: Expose a function from `crates/next-core/src/next_config.rs` that returns a `BoolVc`.